src/visidentifier/visidentifier.cpp

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <Poco/JSON/JSONException.h>
+
+#include "log.hpp"
+#include "utils/json.hpp"
+#include "visidentifier.hpp"
+#include "vismessage.hpp"
+#include "wsexception.hpp"
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+void VISSubscriptions::RegisterSubscription(const std::string& subscriptionId, Handler&& subscriptionHandler)
+{
+    std::lock_guard lock(mMutex);
+
+    LOG_DBG() << "Registred subscription id = " << subscriptionId.c_str();
+
+    mSubscriptionMap[subscriptionId] = std::move(subscriptionHandler);
+}
+
+aos::Error VISSubscriptions::ProcessSubscription(const std::string& subscriptionId, const Poco::Dynamic::Var value)
+{
+    std::lock_guard lock(mMutex);
+
+    const auto it = mSubscriptionMap.find(subscriptionId);
+
+    if (it == mSubscriptionMap.cend()) {
+        LOG_ERR() << "Unexpected subscription id: = " << subscriptionId.c_str();
+
+        return aos::Error::Enum::eNotFound;
+    }
+
+    return it->second(value);
+}
+
+VISIdentifier::VISIdentifier()
+    : mWSClientIsConnected {Poco::Event::EventType::EVENT_MANUALRESET}
+    , mStopHandleSubjectsChangedThread {Poco::Event::EventType::EVENT_AUTORESET}
+
+{
+}
+
+aos::Error VISIdentifier::Init(WSClientItfPtr                    wsClientPtr,
+    std::shared_ptr<aos::iam::identhandler::SubjectsObserverItf> subjectsObserverPtr, const long initTimeoutMiliseconds)
+{
+    if (wsClientPtr == nullptr || subjectsObserverPtr == nullptr) {
+        return aos::ErrorEnum::eInvalidArgument;
+    }
+
+    std::lock_guard lock(mMutex);
+
+    Close();
+
+    mWsClientPtr         = std::move(wsClientPtr);
+    mSubjectsObserverPtr = std::move(subjectsObserverPtr);
```

Comment:
i'm not sure this ok to move from shared_ptr, because from API looks like this should be used someone else

Reply:
It's accepted by value.
So move wan't break anything.